prow/plugins/dog/dog_test.go

```diff
@@ -0,0 +1,206 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dog
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"regexp"
+	"strings"
+	"testing"
+
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/github/fakegithub"
+)
+
+type fakeClowder string
+
+var human = flag.Bool("human", false, "Enable to run additional manual tests")
+var category = flag.String("category", "", "Request a particular category if set")
+
+func (c fakeClowder) readDog() (string, error) {
+	return string(c), nil
+}
+
+func TestRealCat(t *testing.T) {
+	if !*human {
+		t.Skip("Real dogs disabled for automation. Manual users can add --human [--category=foo]")
+	}
+	if cat, err := dogURL.readDog(); err != nil {
+		t.Errorf("Could not read dog from %s: %v", dogURL, err)
+	} else {
+		fmt.Println(cat)
+	}
+}
+
+func TestFormat(t *testing.T) {
```

Comment:
I do not see where you are validating the behavior when the url returns unexpected json

Reply:
done now in `TestHttpResponse`